src/GovUk.Education.ExploreEducationStatistics.Publisher/Services/NotificationsService.cs

```diff
@@ -57,22 +55,32 @@ public async Task NotifySubscribersIfApplicable(params Guid[] releaseIds)
             }
         }
 
-        private IEnumerable<Publication> GetPublications(IEnumerable<Guid> releaseIds)
+        private async Task<PublicationNotificationMessage> BuildPublicationNotificationMessage(Release release)
         {
-            return _context.Releases
-                .AsNoTracking()
-                .Where(release => releaseIds.Contains(release.Id))
-                .Select(release => release.Publication)
-                .Distinct();
-        }
+            await _context.Entry(release)
+                .Reference(r => r.Publication)
+                .LoadAsync();
+
+            await _context.Entry(release)
+                .Collection(r => r.Updates)
+                .LoadAsync();
+
+            var latestUpdateNote = release.Updates
+                .OrderBy(u => u.Created)
+                .LastOrDefault();
+
+            var latestUpdateNoteReason =
```

Comment:
Can there ever be no update note provided?

It's just that this code potentially allows for a scenario where a subscriber is sent an email

```
The update note for this change is:
  * No update note provided
```

or even worse if the update note is the empty string

```
The update note for this change is:
  * 
```

If we ever allow no update note I think it would be better to make the `UpdateNote` field of the message optional, and handle the update note and the surrounding wording as optional content in Gov.Uk notify. See [Optional Content](https://www.notifications.service.gov.uk/using-notify/guidance/edit-and-format-messages#optional-content).

If we don't ever allow null then you will already know I'm not a fan of this defensive coding where it's not necessary and would use `Last` rather than `LastOrDefault` to cause an exception so we are alerted to investigate what is allowing no update note.

Reply:
Discussed this with Ben and Cam on Slack. For now, I'm going to leave the `The update note for this change is:
  * No update note provided` text, as amendments should always have a new update note, so by changing this we'd be protecting against a possibility that should never happen.

I have changed the `LastOrDefault` to `Last`, but that DB query only runs if it is an amendment being published. I've also added comments to the code clarifying that the update_note is only used in the amendment email template, not in the original release email template. This means that if someone tries to publish an amendment 

There is a question on whether it is better for the service to blow up if it expects an update note and one isn't found, or if it should continue. Not sure what others think about this, both in this instance and across the codebase generally? 

On the one hand, we want to avoid having lots of bugs that we're unaware of, but at the same time we don't want the service to have production issues - perhaps is better to have a production issue sooner rather than later, and have bugs that have made loads of bad data etc. 

Also possible there should be some balance between "make it blow up" and "defensively code" depending on the situation. Maybe in a particular instance the cost of a production issue is so high that making it blow up is a bad idea - even if that risks having a bug that doesn't present itself?